File: app/src/main/java/com/tichuguru/db/PlayerDao.java
package com.tichuguru.db;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import java.util.List;

@Dao
public interface PlayerDao {
    @Query("SELECT * FROM players ORDER BY name")
    List<PlayerEntity> getAll();

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    List<Long> insertAll(List<PlayerEntity> players);

    @Query("DELETE FROM players")
    void deleteAll();
}

File: app/src/main/kotlin/com/tichuguru/db/PlayerDao.kt
package com.tichuguru.db

import androidx.room.Dao
import androidx.room.Query
import androidx.room.Upsert

@Dao
interface PlayerDao {
    @Query("SELECT * FROM players ORDER BY name")
    fun getAll(): List<PlayerEntity>

    @Upsert
    fun upsertAll(players: List<PlayerEntity>): List<Long>
}
